src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1035,32 +1035,43 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
             return FindTargetAttribute(token, AttributeDescription.IsByRefLikeAttribute).HasValue;
         }
 
-        internal bool HasDeprecatedOrExperimentalOrObsoleteAttribute(EntityHandle token, out ObsoleteAttributeData obsoleteData)
+        internal const string byRefLikeMarker = "Types with embedded references are not supported in this version of your compiler.";
+
+        internal ObsoleteAttributeData TryGetDeprecatedOrExperimentalOrObsoleteAttribute(
+            EntityHandle token, 
+            bool ignoreByRefLikeMarker)
         {
             AttributeInfo info;
 
             info = FindTargetAttribute(token, AttributeDescription.DeprecatedAttribute);
             if (info.HasValue)
             {
-                return TryExtractDeprecatedDataFromAttribute(info, out obsoleteData);
+                return TryExtractDeprecatedDataFromAttribute(info);
             }
 
             info = FindTargetAttribute(token, AttributeDescription.ObsoleteAttribute);
             if (info.HasValue)
             {
-                return TryExtractObsoleteDataFromAttribute(info, out obsoleteData);
+                ObsoleteAttributeData obsoleteData = TryExtractObsoleteDataFromAttribute(info);
+                if (obsoleteData != null &&
+                    ignoreByRefLikeMarker &&
+                    obsoleteData.Message == byRefLikeMarker)
```

Comment:
yes "==" is the simplest byte-to-byte kind of string compare. 

---
In reply to: [123848279](https://github.com/dotnet/roslyn/pull/20430#discussion_r123848279) [](ancestors = 123848279)